docs/develop/financial-accounting.md

```diff
@@ -0,0 +1,187 @@
+bookkeeping systems use **debits and credits**.
+
+The two entries that give "double-entry bookkeeping" its name are the debit and the credit: every
+transaction has at least one debit and at least one credit. (Note that for efficiency's sake,
+TigerBeetle `Transfer`s consist of exactly one debit and one credit. These can be composed into more
+complex [multi-debit, multi-credit transfers](./recipes/multi-debit-credit-transfers.md).) Which
+entry is the debit and which is the credit?
+
+When our example loan increases the assets and liabilities, we need to assign each of these entries
+to either be a debit or a credit. At some level, this is completely arbitrary. For clarity,
+accountants have used the same standards for hundreds of years:
+
+### How Debits and Credits Increase or Decrease Account Balances
+
+- **Assets and expenses are increased with debits, decreased with credits**
+- **Liabilities, equity, and revenue are increased with credits, decreased with debits**
+
+Or, in a table form:
+
+|             | Debit | Credit |
+| ----------- | ----- | ------ |
+| Assets      | +     | -      |
+| Liabilities | -     | +      |
+| Equity      | -     | +      |
+| Revenue     | -     | +      |
+| Expenses    | +     | -      |
+
+From the perspective of our example bank:
+
+- You taking out a loan debits (increases) their loan assets and credits (increases) their bank
+  account balance liabilities.
+- You paying off the loan debits (decreases) their bank account balance liabilities and credits
+  (decreases) their loan assets.
+- You depositing cash debits (increases) their cash assets and credits (increases) their bank
+  account balance liabilities.
+- You withdrawing cash debits (decreases) their bank account balance liabilities and credits
+  (decreases) their cash assets.
+
+Note that accounting conventions also always write the debits first, to represent the funds flowing
+_from_ the debited account _to_ the credited account.
+
+If this seems arbitrary and confusing, we understand! It's a convention, just like how most
+programmers need to learn zero-based array indexing and then at some point it becomes second nature.
+
+## Takeaways
+
+- Accounts are categorized into types. The 5 main types are assets, liabilities, equity, revenue,
+  and expenses.
+- Depending on the type of account, an increase is recorded as either a debit or a credit.
+- All transfers consist of two entries, a debit and a credit. Double-entry bookkeeping ensures that
+  all funds come from somewhere and go somewhere.
+
+When you get started using TigerBeetle, we would recommend writing a list of all the types of
+accounts in your system that you can think of. Then, think about whether, from the perspective of
+your business, each account represents an asset, liability, equity, revenue, or expense. That
+determines whether the given type of account is increased with a debit or a credit.
```

Comment:
Would be great to motivate why knowing the type of an account is more important (i.e. preserves more information) than what some APIs refer to as knowing "the normal balance" (is the normal balance for the account debit or credit). Again, the reason is that the type also tells you the normal balance, but the normal balance alone doesn't tell you the type (e.g. an account could have a normal balance on the debit side, but you still don't know whether it's an asset or expense).